src/Akka.CQRS.Pricing.Service/Program.cs

```diff
@@ -48,58 +53,80 @@ public static async Task<int> Main(string[] args)
             await Task.Delay(TimeSpan.FromSeconds(15));
             
             var config = await File.ReadAllTextAsync("app.conf");
-            var conf = ConfigurationFactory.ParseString(config)
-                .WithFallback(GetSqlHocon(sqlConnectionString, sqlProviderName))
-                .WithFallback(OpsConfig.GetOpsConfig())
-                .WithFallback(ClusterSharding.DefaultConfig())
-                .WithFallback(DistributedPubSub.DefaultConfig())
-                .WithFallback(SqlPersistence.DefaultConfiguration);
-
-            var actorSystem = ActorSystem.Create("AkkaPricing", conf.BootstrapFromDocker());
-            var readJournal = actorSystem.ReadJournalFor<SqlReadJournal>(SqlReadJournal.Identifier);
-            var priceViewMaster = actorSystem.ActorOf(Props.Create(() => new PriceViewMaster()), "prices");
-
-            Cluster.Cluster.Get(actorSystem).RegisterOnMemberUp(() =>
+            using var host = new HostBuilder()
+            .ConfigureServices((hostContext, services) =>
+            {
+
+                services.AddAkka("AkkaPricing", options =>
+                {
+                    // Add HOCON configuration from Docker
+                    var conf = ConfigurationFactory.ParseString(config)                 
+                    .WithFallback(GetSqlHocon(sqlConnectionString, sqlProviderName))                
+                    .WithFallback(OpsConfig.GetOpsConfig())                 
+                    .WithFallback(ClusterSharding.DefaultConfig())                 
+                    .WithFallback(DistributedPubSub.DefaultConfig())                 
+                    .WithFallback(SqlPersistence.DefaultConfiguration);
+                    options.AddHocon(conf.BootstrapFromDocker(), HoconAddMode.Prepend)
+                    .WithActors((system, registry) =>
+                    {
+                        var priceViewMaster = system.ActorOf(Props.Create(() => new PriceViewMaster()), "prices");
+                        registry.Register<PriceViewMaster>(priceViewMaster);
+                        // used to seed pricing data
+                        var readJournal = system.ReadJournalFor<SqlReadJournal>(SqlReadJournal.Identifier);
+                        Cluster.Cluster.Get(system).RegisterOnMemberUp(() =>
+                        {
+                            var sharding = ClusterSharding.Get(system);
+
+                            var shardRegion = sharding.Start("priceAggregator",
+                                s => Props.Create(() => new MatchAggregator(s, readJournal)),
+                                ClusterShardingSettings.Create(system),
+                                new StockShardMsgRouter());
+
+                            // used to seed pricing data
+                            var singleton = ClusterSingletonManager.Props(

```

Comment:
This can also be extracted to use `.WithSingleton<TKey>` hosting extension instead.